### ISSUE TITLE
ARM32: JIT seemingly reads wrong field

### ISSUE THREAD

**Issue by jakobbotsch** [dotnet/runtime#57065]
```csharp
// Generated by Fuzzlyn v1.2 on 2021-08-06 19:31:41
// Run on .NET 6.0.0-dev on Arm Linux
// Seed: 5500224797583134883
// Reduced from 62.7 KiB to 0.5 KiB in 00:01:25
// Debug: Prints 0 line(s)
// Release: Prints 1 line(s)
struct S0
{
    public int F0;
    public short F1;
    public int F2;
    public bool F3;
}

class C0
{
    public ushort F1;
}

public class Program
{
    static S0[, ] s_1 = new S0[, ]{{new S0()}};
    static C0 s_4 = new C0();
    public static void Main()
    {
        S0 vr0 = default(S0);
        long vr1 = vr0.F0++;
        s_1[0, 0] = vr0;
        if (s_1[0, 0].F3)
        {
            ushort vr2 = s_4.F1;
            System.Console.WriteLine(vr2);
        }
    }
}
```

**Comment by sandreenko** [comment]
it appeared to be a different issue, not a VN one. Will take a look tomorrow.

**Comment by sandreenko** [comment]
It was fixed by @jakobbotsch in #57279.
